smithy-model/src/main/java/software/amazon/smithy/model/shapes/SmithyIdlModelSerializer.java

```diff
@@ -145,15 +145,39 @@ private String serialize(Model fullModel, Collection<Shape> shapes) {
         SmithyCodeWriter codeWriter = new SmithyCodeWriter(namespace, fullModel);
         NodeSerializer nodeSerializer = new NodeSerializer(codeWriter, fullModel);
 
-        ShapeSerializer shapeSerializer = new ShapeSerializer(codeWriter, nodeSerializer, traitFilter, fullModel);
+        Set<ShapeId> inlineableShapes = getInlineableShapes(fullModel, shapes);
+        ShapeSerializer shapeSerializer = new ShapeSerializer(
+                codeWriter, nodeSerializer, traitFilter, fullModel, inlineableShapes);
         shapes.stream()
                 .filter(FunctionalUtils.not(Shape::isMemberShape))
+                .filter(shape -> !inlineableShapes.contains(shape.getId()))
                 .sorted(new ShapeComparator())
                 .forEach(shape -> shape.accept(shapeSerializer));
 
         return serializeHeader(fullModel, namespace) + codeWriter.toString();
     }
 
+    private Set<ShapeId> getInlineableShapes(Model fullModel, Collection<Shape> shapes) {
```

Comment:
We'll need to modify this once #980 lands

Reply:
Yeah, we'll also need to add the input and output shapes by default, so we can change it as part of that.